### PR TITLE
Protocol 12.20

### PR DESCRIPTION
**Pull request by FakeShinoda**
- You'll need to change accountname to email in your login.php
- This PR will only work with client 12.20

- Credits to SaiyansKing & R1Ck
- _Thanks to majesty for login by email_

### CODE REVIEW

**Comment by majestyotbr** [comment]
Client 12.20:
https://mega.nz/file/7ip32B5T#uHrJwJWiMu6fLNtNqPHDdMdt2P8BUwmWMnCqYZnjIDg

Login.php to test:
GesiorAAC: https://github.com/opentibiabr/gesioraac-tibia12-login/releases/tag/1.1
MyAAC: https://github.com/opentibiabr/myaac-tibia12-login/releases/tag/1.4

**Comment by Znote** [comment]
Tested this PR, and fixed Email login support on latest version of Znote AAC. 
https://github.com/Znote/ZnoteAAC/commit/3c8b1eb0aad440fe2ea7bf6087aafa6eaa3687e9

Works. If anyone has issues connecting to game server, put this script into your data/scripts/ folder:
znote_login.lua
https://raw.githubusercontent.com/Znote/ZnoteAAC/3c8b1eb0aad440fe2ea7bf6087aafa6eaa3687e9/LUA/TFS_10/revscriptsys/znote_login.lua

And restart server.

**Comment by majestyotbr** [comment]
Thanks @Znote!

**Comment by MajestyOTBR** [comment]
Can only be merged with PR 1381 (Gamestore 12.20).

File: src/player.h [hunk 1]
```diff
@@ -1598,8 +1598,7 @@ class Player final : public Creature, public Cylinder
 		BlockType_t lastAttackBlockType = BLOCK_NONE;
 		tradestate_t tradeState = TRADE_NONE;
 		fightMode_t fightMode = FIGHTMODE_ATTACK;
-		account::AccountType accountType =
-                                    account::AccountType::ACCOUNT_TYPE_NORMAL;
+		account::AccountType accountType = account::AccountType::ACCOUNT_TYPE_NORMAL;
```
Comment (dudantas) [inline src/player.h]: Can you return this to what it was before?

File: src/protocolgame.cpp [hunk 2]
```diff
@@ -792,16 +792,25 @@ void ProtocolGame::parseSetOutfit(NetworkMessage& msg)
...
+			newOutfit.lookMount = msg.get<uint16_t>();
+		} else if (outfitType == 1) {
+			//This value probably has something to do with try outfit variable inside outfit window dialog
+			//if try outfit is set to 2 it expects uint32_t value after mounted and disable mounts from outfit window dialog
+			newOutfit.lookMount = 0;
+			msg.get<uint32_t>();
+		}
 		addGameTask(&Game::playerChangeOutfit, player->getID(), newOutfit);
-	}
+  	}
```
Comment (dudantas) [inline src/protocolgame.cpp]: And this indentation

File: src/tile.cpp [hunk 3]
```diff
@@ -1647,3 +1647,22 @@ Item* Tile::getUseItem(int32_t index) const
...
+    	for (Item* item : *items) {
+      		const ItemType& it = Item::items[item->getID()];
+      		if (it.isDoor()) {
+        		return item;
+     		}
+    	}
+  	}
+
+	return nullptr;
+}
```
Comment (dudantas) [inline src/tile.cpp]: Please keep the last line too

**Review by dudantas — APPROVED** [review]
LGTM

Just make the small changes I asked for, the rest seems ok

**Comment by cadugtx** [comment]
doesnt work, using gesior, "connection failed"

**Comment by MajestyOTBR** [comment]
Of course it works, configure it correctly. This is not the place to request support.

**Pull request merged**